modules/host/info.go

```diff
@@ -57,21 +104,24 @@ func (h *Host) RPCMetrics() modules.HostRPCMetrics {
 func (h *Host) SetSettings(settings modules.HostSettings) error {
 	h.mu.Lock()
 	defer h.mu.Unlock()
+	// Resource lock is grabbed for the save function.
 	h.resourceLock.RLock()
 	defer h.resourceLock.RUnlock()
 	if h.closed {
 		return errHostClosed
 	}
 
-	// Check that the unlock hash was not changed.
+	// Check that none of the illegal fields have been modified.
```

Comment:
perhaps. It would be nice to be able to add internal-only fields such as the auto-pricing bools.